yt_idefix/data_structures.py

```diff
@@ -724,3 +816,290 @@ def _validate_units_override_keys(cls, units_override):
             )
 
         super(cls, cls)._validate_units_override_keys(units_override)
+
+
+class PlutoXdmfDataset(PlutoVtkDataset):
+    _index_class = PlutoXdmfHierarchy
+    _field_info_class = PlutoXdmfFields
+    _dataset_type = "pluto-xdmf"
+    _version_regexp = re.compile(r"\d+\.\d+\.?\d*[-\w+]*")
+    _required_header_keyword = "PLUTOXdmf"
+    _default_definitions_header = "definitions.h"
+    _default_inifile = "pluto.ini"
+    _field_offset_index = None
+
+    def _parse_parameter_file(self):
```

Comment:
two notes here:
- `self.parameter_filename` is a historical alias and should be replaced with `self.filename`
- a shortcut `self.directory` can be used to avoid repeating `os.path.dirname(self.parameter_filename)`

it would make this function easier to read to leverage these

Reply:
I see that ```self.parameter_filename``` is used in many parts of the code and I'm a bit confused on where to replace ```self.parameter_filename``` with ```self.filename```.
I have used the ```self.directory``` shortcut as you suggested.